### PR TITLE
initial plugin upload

### PR DESCRIPTION
**Pull request by jgrieb**
the first version of the qgis shogun editor plugin

### CODE REVIEW

**Comment by KaiVolland** [comment]
Looks good to me. Maybe @marcjansen want to have a quick look at it again?

File: src/shoguneditor/__init__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/***************************************************************************
+ ShogunEditor
+                                 A QGIS plugin to connect with a Shogun
+                                 GIS client instance on a remote or local server
+                                 and edit it's content from QGIS
+
+                             -------------------
+        begin                : 2018-05-11
+        copyright            : (C) 2018 by terrestris GmbH & CO. KG
+        email                : jgrieb (at) terretris.de
```
Comment (marcjansen) [inline src/shoguneditor/__init__.py]: typo

File: src/shoguneditor/__init__.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+'''
+/***************************************************************************
+ ShogunEditor
+                                 A QGIS plugin to connect with a Shogun
+                                 GIS client instance on a remote or local server
+                                 and edit it's content from QGIS
+
+                             -------------------
+        begin                : 2018-05-11
+        copyright            : (C) 2018 by terrestris GmbH & CO. KG
```
Comment (marcjansen) [inline src/shoguneditor/__init__.py]: Spelling. `Co.`

I would also like to have the info@ email adress in. you can probably add more than one.

File: src/shoguneditor/connection/__init__.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+'''
+(c) 2018 Terrestris GmbH & CO. KG, https://www.terrestris.de/en/
+ This code is licensed under the GPL 2.0 license.
```
Comment (marcjansen) [inline src/shoguneditor/connection/__init__.py]: Typos. terrestris is all lowercase, Co. with a lowercase o, please. here and probably elsewhere.

File: src/shoguneditor/connection/shogunressource.py [hunk 4]
```diff
@@ -0,0 +1,463 @@
+# -*- coding: utf-8 -*-
+'''
+(c) 2018 Terrestris GmbH & CO. KG, https://www.terrestris.de/en/
+ This code is licensed under the GPL 2.0 license.
+'''
+
+__author__ = 'Jonas Grieb'
+__date__ = 'Juli 2018'
```
Comment (marcjansen) [inline src/shoguneditor/connection/shogunressource.py]: This should be `July`, right? Here and elsewhere.

File: src/shoguneditor/connection/shogunressource.py [hunk 5]
```diff
@@ -0,0 +1,463 @@
...
+        self.iface = iface
+        if url.endswith('webapp'):
+            url+='/'
+        elif url.endswith('rest/'):
+            url = url[:-5]
+        elif url.endswith('rest'):
+            url = url[:-4]
+
+        if not url.startswith('http://') and not url.startswith('https://'):
+            url = 'http://' + url
```
Comment (marcjansen) [inline src/shoguneditor/connection/shogunressource.py]: Please add a TODO, that we want to test HTTPS first with a Head request, and then only fallback to HTTP at the very end.

**Comment by marcjansen** [comment]
The remaining issues should be tackled separately

**Pull request merged**